tools/get_dicom_files.pl

```diff
@@ -0,0 +1,292 @@
+        push(@{ $filesRef->{$mincFile} }, $dicomFilename);
+    }
+    
+    # Foreach MINC file build a tar file with the set of DICOM files
+    # that were used to produce it (found above)
+    foreach my $file (keys %$filesRef) {
+        # Build file that contains the list of paths of the DICOM files to
+        # extact from the archive
+        my @dicomFilePatterns = map { "*/$_" } @{ $filesRef->{$file} };
+        my($fileBaseName) = $file =~ /\/([^\/]+).mnc$/;
+        my $fileList = "$tmpExtractDir/$fileBaseName.dicom";
+        open(FILE_LIST, ">$fileList") or die "Cannot write file $fileList: $!\n";
+        foreach( @dicomFilePatterns) {
+            print FILE_LIST "$_\n";
+        }
+        close(FILE_LIST);
+        
+        # Extract from the inner tar the DICOMs who names are listed in
+        # $tmpExtractDir/$fileBaseName.dicom
+        my($outSubDir) = $file =~ /_([^_]+_\d+).mnc$/;
+        my $outDir = "$tmpExtractDir/$pscid/$visitLabel/$dateAcquired/$outSubDir";
+        my $cmd = "tar zxf $tmpExtractDir/*.tar.gz"
+                . " --no-recursion "
+                . " --wildcards"
+                . " --files-from=$fileList "
+                . " --absolute-names "
+                . " --transform='s#^.*/#$outDir/#'";
+        print "Extracting DICOM files for $file...";
+        system($cmd) == 0 
+            or die "Failed to extract DICOM files for MINC file $file from archive in $tmpExtractDir: $?\n";
+        print "done.\n";
+        
+        push(@outDirs, "$pscid/$visitLabel/$dateAcquired/$outSubDir");
+    }
+}
+
+if(!@outDirs) {
+    print "No DICOM files match the specified criterias specifed via -t and/or -n. No .tar.gz file created.\n";
+} else {
+    # Make a big tar of all the extracted DICOMs that match the patient name patterns
+    # and scan type name patterns
+    my $outTarFile = "$outTarBasename.tar.gz";
+    my $outDirsFile = "$tmpExtractDir/out_dirs.$$.txt";
+    open(OUT_DIRS_FILE, ">$outDirsFile") or die "Cannot write to file $outDirsFile: $!\n";
+    foreach(@outDirs) {
+        print OUT_DIRS_FILE "$_\n";
+    }
+    close(OUT_DIRS_FILE);
+
+    print "Writing $outTarFile...";
+    my $cmd = "tar zcf $outTarFile -C $tmpExtractDir --files-from=$outDirsFile";
+    system($cmd) == 0 or die "Failed to archive DICOM files in $outTarFile: $?\n";
+    print "done\n";
+}
+
+exit $NeuroDB::ExitCodes::SUCCESS;
+
+#-----------------------------------------------------#
+# This will erase the tmp extract dir when the script #
+# terminates (either normally or abnormally)          #
+#-----------------------------------------------------#
+END { &rmTmpExtractDir() }
+
+#==========================#
+#                          #
+#       SUBROUTINES        #
+#                          #
+#==========================#
+
+=pod
+
+=head3 rmTmpExtractDir()
+
+Function that deletes directory C<$tmpExtractDir> and all its contents if 
+variable C<$tmpExtractDir> is defined. 
+
+=cut
+sub rmTmpExtractDir 
```

Comment:
@nicolasbrossard Could this function be moved to one of the NeuroDB library and called by every script using a tmp directory? I am basically rewriting your function in my tool script to deface. 

If you think it is a good idea, I'll create an issue on Github for harmonisation on how we handle the tmp directories in LORIS-MRI.

Reply:
Actually, I just noticed that you could simply to something like that to create tmp directories and remove them when the script is interrupted:

```
use File::Temp qw/ tempfile tempdir /;

my $tmpdir = &tempdir( "get-dicom-files-XXXXXXXX", TMPDIR => 1, CLEANUP => 1 );
```

Reply:
I used your `File::Temp` package. Much cleaner!